dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -47,9 +49,21 @@
   private final ClassDumper classDumper;
   private final ImmutableMap<Path, SymbolReferenceSet> jarToSymbols;
   private final ClassReferenceGraph classReferenceGraph;
+  private final ImmutableMap<Path, Artifact> pathToArtifact;
 
   public static LinkageChecker create(List<Path> jarPaths, Iterable<Path> entryPoints)
       throws IOException {
+    return create(jarPaths, entryPoints, null);
+  }
+
+  public static LinkageChecker create(
+      ImmutableMap<Path, Artifact> pathToArtifact, Iterable<Path> entryPoints) throws IOException {
+    return create(ImmutableList.copyOf(pathToArtifact.keySet()), entryPoints, pathToArtifact);
+  }
+
+  public static LinkageChecker create(
+      List<Path> jarPaths, Iterable<Path> entryPoints, ImmutableMap<Path, Artifact> pathToArtifact)
```

Comment:
e.g. instead of passing lists of paths and maps of paths to artifacts perhaps we need a type that includes both a path and an artifact. Not to fix in this PR, but let's think about that.

Reply:
Right. I was not sure about `pathToArtifact` either. Artifact has `getFile()` and File has `toPath()`. Let me think about better data structure.